gregBorchers/HW6Problem3tarot.js

```diff
@@ -0,0 +1,329 @@
+/**
+---
+**3) Cards Finale** _[25%]_
+
+**a)**
+Rewrite your _makeCard_ factory as a constructor _Card_.  
+You should define your constructor and any supplementary code 
+inside an IIFE and export it to a global variable _Card_.
+
+_Card_ should initialize card objects which have the same four 
+instance methods as before (`rank()`, `suit()`, `color()`, and `name()`), 
+plus another method `isValid()`, which returns _true_ if the card has a valid id.
+Implement each of those instance methods using _Card_'s 
+prototype instead of linking them to each card.  
+Make sure each method uses `this` to refer to the card instance.
+
+_Card_ should also have four class methods which are called through the constructor:
+
+- `isCard(card)` returns true if card is a valid instance of the class _Card_;
+- `numCards()` returns the number of cards in a full deck;
+- `rankNames()` returns an array of all rank names;
+- `suitNames()` returns an array of all suit names.
+
+For both name arrays: omit any empty "padding" strings, and 
+protect the internal name arrays by returning only a copy.
+
+There are multiple ways of organizing your module to satisfy 
+these requirements, but some of the patterns will be much easier to adapt to part (b).  
+Read ahead carefully and study the included [template file](template-tarot.js).
+
+**b)** The Card class will have a subclass TarotCard, 
+another constructor which inherits from it.  
+Tarot cards (Minor only!) are similar to ordinary playing cards, 
+but with four differences:
+
+1. Their suit names are "Cups", "Pentacles", "Swords", and "Wands".
+1. A Jack is called instead a "Knight", and there is 
+one extra rank "Page" between "Ten" and "Knight".  
+Therefore there are 56 total cards.
+
+1. Tarot cards have no color, and should have no such method.
+1. Each tarot card can be oriented normally or upside-down, 
+and has a boolean instance property reflecting that orientation.
+
+The _TarotCard_ constructor is implemented in a separate module, 
+and it has been written completely in the [template](template-tarot.js).  
+You don't need to change anything in that module.  
+But you should write your _Card_ module in such a way that _TarotCard_ 
+can sucessfully inherit from it.  
+An expression like `card = new Card(51)` should make an object with Card behavior, and 
+`tarot = new TarotCard(51)` should make an object with Tarot behavior.
+The two classes should be able to co-exist and pass all of the tests in the template.
+
+*/
+
+
+
+//____________________________________________ POKER CARD
+// This IIFE will define a superclass:
+var Card = (function() {
+
+    // Ctor:
+    function Card(id) {
+            // **** properties of each card            
+            if (true) { //TOFIX how to subclass max card limits
+                this.id = id; 
```

Comment:
If you give each constructor a variable that represents the maximum number of cards, you can then check to see if the current card id is less than that number by checking in with the constructor.